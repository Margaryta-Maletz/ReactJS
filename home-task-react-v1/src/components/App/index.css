.wrapper {
    width: 1200px;
    box-sizing: border-box;
    padding: 20px 60px;
    display: flex;
}
