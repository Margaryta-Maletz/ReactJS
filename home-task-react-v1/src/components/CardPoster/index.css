.card-poster_wrapper {
    width: 322px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    color: white;
    mix-blend-mode: normal;
}

.card-poster_title {
    font-size: 18px;
    line-height: 22px;
    opacity: 0.7;
    margin: 22px 0 8px;
    width: 75%;
}

.card-poster_genre {
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
    margin-block: 0;
    margin: 0 0 50px;
}

.card-poster_release-date {
    width: 66px;
    height: 26px;
    border: 1px solid #555555;
    box-sizing: border-box;
    border-radius: 4px;
    font-size: 14px;
    line-height: 24px;
    text-align: center;
    margin: 22px 0 8px;
}
