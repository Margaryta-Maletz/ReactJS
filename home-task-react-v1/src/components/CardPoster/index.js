import React from 'react';

export const CardPoster = (props) => {
    return (
        <div className='card-poster_wrapper'>
            <img className='card-poster_image' src={ props.cards.imagePath } alt={ props.cards.imageAlt } />
            <h4 className='card-poster_title'>{ props.cards.title }</h4>
            <p className='card-poster_genre'>{ props.cards.genre }</p>
            <p className='card-poster_release-date'>{ props.cards.releaseDate }</p>
        </div>
    )
}
