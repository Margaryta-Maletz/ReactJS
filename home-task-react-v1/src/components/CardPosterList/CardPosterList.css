.card-poster-list_wrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: flex-start;
}
