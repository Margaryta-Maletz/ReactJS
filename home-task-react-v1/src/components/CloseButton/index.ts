export { CloseButton, CloseButtonSize } from './CloseButton';
