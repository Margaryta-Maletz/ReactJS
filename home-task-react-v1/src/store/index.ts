import thunk from 'redux-thunk';
import {configureStore, getDefaultMiddleware, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './slice';

const middleware = getDefaultMiddleware({
    imitableCheck: false,
    serializableCheck: false,
    thunk: true,
});

export const store = configureStore({
    reducer: rootReducer,
    middleware: middleware,
    devTools: process.env.NODE_ENV !== 'production',
});

/*export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;*/
