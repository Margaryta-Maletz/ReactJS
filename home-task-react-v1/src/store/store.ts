import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './reducers';

export const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;
