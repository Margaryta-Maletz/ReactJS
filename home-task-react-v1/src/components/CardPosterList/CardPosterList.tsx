import React from 'react';
import { useSelector } from 'react-redux';
import { CardPoster } from '../CardPoster';
import './CardPosterList.css';
import { IState } from '../../store/types';

export const CardPosterList: React.FC = () => {
    const { movies } = useSelector((state: IState) => state);

    return (
        <div className="card-poster-list_wrapper">
            { movies.map((item) =>
                <CardPoster key={item.id.toString()} { ...item } />
            )}
        </div>
    )
}
