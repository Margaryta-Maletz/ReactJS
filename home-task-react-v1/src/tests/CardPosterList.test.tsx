import React from "react";
import { render } from '@testing-library/react';
import { CardPosterList } from '../components/CardPosterList/';
import {Provider} from "react-redux";
import {store} from "../store/store";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:8080/search/"
    }),
    useHistory: (path: string) => ({
        push: jest.fn(),
        location: {
            pathname: "localhost:8080/search/" + path
        }
    })
}));

describe('make snapshot', () => {
    test('renders', () => {
        const { container } = render(
            <Provider store={store}>
                <CardPosterList />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
