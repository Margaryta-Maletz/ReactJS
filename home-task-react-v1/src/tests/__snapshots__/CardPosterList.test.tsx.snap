// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders 1`] = `
<div>
  <div
    class="card-poster-list_wrapper"
  >
    <div
      class="card-poster_wrapper"
    >
      <img
        alt="Don't miss the climax"
        class="card-poster_image"
        height="455"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
        width="322"
      />
      <div
        class="edit-button"
      >
        <svg
          fill="none"
          height="44"
          viewBox="0 0 44 44"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            filter="url(#filter0_d)"
          >
            <circle
              cx="22"
              cy="22"
              fill="#2A202D"
              r="18"
            />
          </g>
          <circle
            cx="22"
            cy="15"
            fill="white"
            r="2"
          />
          <circle
            cx="22"
            cy="22.5"
            fill="white"
            r="2"
          />
          <circle
            cx="22"
            cy="30"
            fill="white"
            r="2"
          />
          <defs>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="44"
              id="filter0_d"
              width="44"
              x="0"
              y="0"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.196596 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow"
                mode="normal"
                result="shape"
              />
            </filter>
          </defs>
        </svg>
      </div>
      <h4
        class="card-poster_title"
      >
        Fifty Shades Freed
      </h4>
      <p
        class="card-poster_release-date"
      >
        2018
      </p>
      <p
        class="card-poster_genre"
      >
        Drama & Romance
      </p>
    </div>
  </div>
</div>
`;
